### PR TITLE
Work around ambiguity error with mul!

### PR DESCRIPTION
**Pull request by andreasnoack**
An issue on Julia 1.3

### CODE REVIEW

File: src/ToeplitzMatrices.jl [hunk 1]
```diff
@@ -102,6 +102,10 @@ function mul!(y::StridedVector{T}, A::AbstractToeplitz{T}, x::StridedVector, α:
         return y
     end
 end
+# Avoid ambiguity error
+mul!(y::StridedVector{T}, A::AbstractToeplitz{T}, x::StridedVector, α::T, β::T) where {T<:Number} =
```
Comment (JeffBezanson) [inline src/ToeplitzMatrices.jl]: Could `T<:Number` just be added to the above definition instead?

Comment (andreasnoack) [reply src/ToeplitzMatrices.jl]: I'm not completely sure that the elements are always `Number`s (although it's probably unlikely) which was why I added a separate method.

Comment (dlfivefifty) [reply src/ToeplitzMatrices.jl]: Is there a reason they are specialised to `StridedVector`? The code looks like it should work regardless. (If you are adding the unnecessary restriction to `StridedVector`, might as well go all the way and do `StridedVector{<:Number}`).

Comment (andreasnoack) [reply src/ToeplitzMatrices.jl]: It's been a long time so I don't remember the details but when I wrote these, I only had floats in mind so if you don't rely on non-Numbers here then we can just restrict it and make a new minor release.

**Pull request merged**